lib/services/s3.js

```diff
@@ -436,13 +459,41 @@ AWS.util.update(AWS.S3.prototype, {
     }
   },
 
+  /**
+   * @api private
+   * @param request
+   */
+  disableBodySigning: function disableBodySigning(request) {
+    // Only file streams are supported (content-md5 still calculated)
+    request.service.validateStreamBody(request.httpRequest.body || '');
+    var headers = request.httpRequest.headers;
+    // Add the header to anything that isn't a presigned url, unless that presigned url had a body defined
+    if (headers.hasOwnProperty('X-Amz-Content-Sha256') || !headers.hasOwnProperty('presigned-expires')) {
+      headers['X-Amz-Content-Sha256'] = 'UNSIGNED-PAYLOAD';
```

Comment:
Is there a reason to override the `X-Amz-Content-Sha256` header if it already exists? If the reason to disable body signing is to prevent expensive computations that could slow down the browser, then what is the benefit of overriding it when it has already been computed? The only place in the SDK that adds the `X-Amz-Content-Sha256` header is in the `AWS.EventListeners.Core.COMPUTE_SHA256` listener, which is already removed if `shouldDisableBodySigning()` returns true.

If a customer decides to compute the Sha256 on their own and adds it to the header, it would seem like surprising behavior that their own header gets overridden, especially if they did not explicitly set `s3DisableBodySigning` to false and did not realize that the SDK defaults that option to true.